src/libsyntax/attr.rs

```diff
@@ -308,6 +315,10 @@ impl MetaItem {
     pub fn is_meta_item_list(&self) -> bool {
         self.meta_item_list().is_some()
     }
+
+    pub fn is_scoped(&self) -> bool {
```

Comment:
You could make this return an option of the first segment and then use that segment in the `is_known_lint_tool` function (and maybe to use a more fine-grained span for the diagnostic)

Reply:
Yeah good idea!